dsc/workflows/opencourseware.py

```diff
@@ -13,6 +14,234 @@
 logger = logging.getLogger(__name__)
 
 
+class OpenCourseWareTransformer:
+    """Transformer for OpenCourseWare (OCW) source metadata."""
+
+    fields: Iterable[str] = [
+        # fields with derived values
+        "dc_title",
+        "dc_date_issued",
+        "dc_description_abstract",
+        "dc_contributor_author",
+        "creativework_learningresourcetype",
+        "dc_subject",
+        "dc_identifier_other",
+        "dc_coverage_temporal",
+        "dc_audience_educationlevel",
+        # fields with static values
+        "dc_type",
+        "dc_rights",
+        "dc_rights_uri",
+        "dc_language_iso",
+    ]
+
+    @classmethod
+    def transform(cls, source_metadata: dict) -> dict:
+        """Transform source metadata."""
+        transformed_metadata = {}
+        for field in cls.fields:
+            field_method = getattr(cls, field)
+            formatted_field_name = field.replace("_", ".")
+
+            # check if 'source_metadata' is in signature
+            signature = inspect.signature(field_method)
+            if "source_metadata" in signature.parameters:
+                transformed_metadata[formatted_field_name] = field_method(source_metadata)
+            else:
+                transformed_metadata[formatted_field_name] = field_method()
```

Comment:
While this is _really_ cool, I wonder if it's necessary?  If we just pass `source_metadata` always, and methods can choose to ignore it, what's the harm?  Methods could have a leading underscore in the signature like this to avoid linting errors and indirectly suggest to developers its not used:

```python
def i_dont_need_metadata(_source_metadata:dict):
    ...
```

Reply:
I'm going to stick to it being 🆒 ! On a more serious note, I think the added code doesn't feel overwhelming or complex. 🤔 It's also the first time I've worked with the `inspect` module. We can revisit this `transform` method in the event we need to create another 'transformer' class.